oauth-authentication-extension/src/browser/oauth-window.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+/**
+ * Popup window to redirect user to OAuth provider site for authentication|authorization.
+ * It is expected that authentication Url contains 'redirect_after_login' query parameter.
+ * In this case popup window will be closed automatically when OAuth provider site redirects user after login to application homepage. 
+ */
+export class OAuthWindow {
+    private popup: Window | null;
+    private popupTrackingId: number | undefined;
+    private closeHandler?: () => void;
+
+    constructor(readonly authenticationUrl: string, readonly width?: number, readonly height?: number) {
+        this.popup = null;
+        this.popupTrackingId = undefined;
+    }
+
+    open(closeHandler?: () => void) {
+        this.closeHandler = closeHandler;
+
+        let width = this.width ? this.width : window.innerWidth - Math.round(window.innerWidth / 3);
```

Comment:
A lot of variables within the PR are declared as `let` but they're actually constants. Do you think they should be declared as `const`?

Reply:
thank you, Artem! I will revise it.